File: src/leetcode/python/easy/IsomorphicStrings_205.py
class IsomorphicStrings_205(object):
    def isIsomorphic(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        m = dict()
        rm = dict()
        for i in range(len(s)):
            l, r = s[i], t[i]
            if l not in m:
                if r in rm:
                    return False
                m[l] = r
                rm[r] = l
            else:
                if m[l] != r:
                    return False
        return True

File: src/leetcode/java/IsomorphicStrings_205.java
package leetcode.java;

public class IsomorphicStrings_205 {
  public boolean isIsomorphic(String s, String t) {
    int[] map = new int[256];
    int[] reverse = new int[256];
    char[] schar = s.toCharArray();
    char[] tchar = t.toCharArray();
    char sc, tc;
    for (int i = 0; i < schar.length; i++) {
      sc = schar[i];
      tc = tchar[i];
      if (map[sc] == 0) {
        if (reverse[tc] != 0) return false;
        map[sc] = tc;
        reverse[tc] = sc;
      } else {
        if (map[sc] != tc) return false;
      }
    }
    return true;
  }
}
